Update to support SDK 2.0

bennettp123 opened the pull request:
Hi there,

I'm sending you this pull request because I forked your tea_timer app and updated it for SDK 2.0. I'm sending this pull request in case you wanted to merge it into your own codebase.

Note that I've also made a few other changes:
- Larger font
- Swapped up and down buttons (up->reduce time, down->increase time; this matches the next/prev behaviour in builtin music app)
- Add ability to stop & start timer
- Snap timer changes to nearest 30 seconds
- Changed default time to three minutes

I've included these in the pull request, but if you want any of them removed, let me know and I'll revert them and resubmit the pull request.

Alternatively, feel free to ignore me and I won't bother you again. :)

Thanks in advance,
Bennett.

bennettp123 committed:
Changed default time to three minutes

bennettp123 committed:
Added ability to toggle (start/stop) the timer

renamed start_timer() function to toggle_timer()

bennettp123 committed:
Swapped up and down buttons

For some reason, pressing down to increment seems more natural to me.

bennettp123 committed:
Larger font

bennettp123 committed:
When changing time, snap to 30 seconds

bennettp123 committed:
Updated for SDK 2.0.

mfollett commented:
Thanks, I haven't looked at SDK 2.0 yet. I'll look at this by the end of the weekend and merge it.

mfollett commented on appinfo.json:
```diff
@@ -0,0 +1,18 @@
+        ""
+    ],
+    "shortName": "tea_timer",
+    "companyName": "bennettp123",
```
Could you make the companyName just an empty string?

bennettp123 replied:
Done

mfollett commented on src/display.c:
```diff
@@ -1,24 +1,29 @@
+//#include <time.h> //already defined in <pebble.h>!
```
Could you remove the commented out code in this file? here

mfollett commented on src/display.c:
```diff
@@ -32,27 +37,40 @@ void alert() {
+	text_layer_set_text_color(timer, GColorWhite);
+	text_layer_set_background_color(timer, GColorClear);
+	//text_layer_set_text_alignment(&timer, GTextAlignmentCenter);
+	//text_layer_set_font(&timer, fonts_get_system_font(FONT_KEY_GOTHAM_30_BLACK));
```
here

bennettp123 committed:
Fixed repeating increase/decrease of timer

Also fixed click handlers to support window_single_repeating_click_subscribe properly.

bennettp123 committed:
Renamed back to "Tea Timer".

mfollett commented on appinfo.json:
```diff
@@ -0,0 +1,18 @@
+    "uuid": "081b46ac-5530-413c-b82e-a192fb698838",
+    "appKeys": {},
+    "longName": "Tea Timer",
+    "versionCode": 128408,
```
This causes a build error. Can you set this to a lower number (1 is fine, or whatever)?

bennettp123 replied:
Done.

bennettp123 committed:
Removed unneeded commented-out code

bennettp123 committed:
Update appinfo.json

mfollett commented:
This is awesome, thanks for updating this.

I realized I don't have a license file in the code. Do you have any objections with this being licensed [BSD 3-clause](http://www.tldrlegal.com/license/bsd-3-clause-license-%28revised%29)?

bennettp123 commented:
Urgh I'm so new to github. Sorry about the close & reopen! I'm learning heaps though. I now know why the recommended workflow is branch-commit-merge! :)

No problems about the update; I found tea_timer it a few months ago and have been using once or twice a day ever since; so naturally, it was the first thing I updated when upgrading to 2.0. :)

I have no objections to the BSD 3-clause license.

mfollett commented:
Great, thanks again for your help!
